src/WaitosaurBase.sol

```diff
@@ -87,57 +79,57 @@ abstract contract WaitosaurBase is
 
     function _setRoles(address locker, address unlocker) internal {
         WaitosaurAccess storage r = _getRoles();
-        require(
-            locker != address(0) && unlocker != address(0),
-            InvalidRolesAddresses()
-        );
+        require(locker != address(0) && unlocker != address(0), InvalidRolesAddresses());
 
         r.locker = locker;
         r.unlocker = unlocker;
 
         emit RolesUpdated(r);
     }
 
-    function updateRoles(
-        address newLocker,
-        address newUnlocker
-    ) public onlyOwner {
+    function updateRoles(address newLocker, address newUnlocker) public onlyOwner {
         _setRoles(newLocker, newUnlocker);
     }
 
-    function _lockBase(
-        uint256 amount
-    ) internal returns (WaitosaurState storage state) {
+    function _lockBase(uint256 amount, uint256 initialOracleBalance) internal returns (WaitosaurState storage state) {
         if (amount == 0) revert AmountZero();
         state = _getState();
         if (state.lockedAmount != 0) revert AlreadyLocked();
         state.lockedAmount = amount;
         state.lastLocked = block.timestamp;
+        state.initialOracleBalance = initialOracleBalance;
 
         emit Locked(amount);
     }
 
-    function lock(uint256 amount) public {
+    function lock(uint256 amount) public virtual {
         WaitosaurAccess storage roles = _getRoles();
         if (_msgSender() != roles.locker && _msgSender() != owner()) {
             revert Unauthorized();
         }
-        _lockBase(amount);
+        uint256 initialBalance = _getInitialOracleBalance();
+        _lockBase(amount, initialBalance);
+    }
+
+    /// @notice Override this to provide initial oracle balance before locking
+    function _getInitialOracleBalance() internal view virtual returns (uint256) {
+        return 0;
```

Comment:
I think it's better to revert here to be extra sure that we don't forget to override this function in child contracts.